File: SchoolMenu/SchoolMenu/ClientApp/src/app/models/Meal.ts
export class Meal {
    id: string;
    incomePrice: number;
    outcomePrice: number;
    name: string;
    recipe: string;
    products: string;

    public constructor(id: string, incomePrice: number, outcomePrice: number, name: string, recipe: string, products: string) {
        this.id = id;
        this.incomePrice = incomePrice;
        this.outcomePrice = outcomePrice;
        this.name = name;
        this.recipe = recipe;
        this.products = products;
    }
};

export class MealAdd {
    name: string;
    price: number;
    recipe: string;
    menuId: string;
    productIds: string[];

    public constructor(name:string,price:number, recipe: string, menuId: string, productIds: string[]) {
        this.name = name;
        this.price = price;
        this.recipe = recipe;
        this.menuId = menuId;
        this.productIds = productIds;
    }
};

File: SchoolMenu/SchoolMenu.DAL/Entities/Meal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SchoolMenu.DAL.Entities
{
    public class Meal
    {
        [Key]
        public Guid Id { get; set; }
        [MaxLength(128)]
        public string Name { get; set; }
        public decimal Price { get; set; }
        public string Recipe { get; set; }
        public ICollection<DayMenu> DayMenus { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}
